PCSX2/pcsx2
# OSD: Move hardcore notification from toast to OSD

### Description of Changes

This PR moves the Hardcore mode notification from toast to the OSD.

Before:
![Screenshot_20240724_153930](https://github.com/user-attachments/assets/58878d99-0010-4f61-b1d0-1347188937f3)


After:
![Screenshot_20240724_155918](https://github.com/user-attachments/assets/52f10897-61fc-4b17-aac2-562baac02e48)


### Rationale behind Changes

Makes it more obvious of its existence and makes it easier to read.

### Suggested Testing Steps

See if the notification shows up properly.

#### Steps to test
1. Enable Hardcore Mode
2. Run a game that has no achievements
3. Observe the notification on the top left
